transaction/src/manifest/lexer.rs

```diff
@@ -372,15 +373,12 @@ impl Lexer {
             "Struct" => Ok(TokenKind::Struct),
             "Enum" => Ok(TokenKind::Enum),
             "Option" => Ok(TokenKind::Option),
-            "Box" => Ok(TokenKind::Box),
             "Array" => Ok(TokenKind::Array),
             "Tuple" => Ok(TokenKind::Tuple),
             "Result" => Ok(TokenKind::Result),
-            "Vec" => Ok(TokenKind::Vec),
-            "TreeSet" => Ok(TokenKind::TreeSet),
-            "TreeMap" => Ok(TokenKind::TreeMap),
-            "HashSet" => Ok(TokenKind::HashSet),
-            "HashMap" => Ok(TokenKind::HashMap),
+            "List" => Ok(TokenKind::List),
+            "Set" => Ok(TokenKind::Set),
+            "Map" => Ok(TokenKind::Map),
```

Comment:
I really like this change. Just wondering, would we like to keep aliases in the transaction manifest Lexer to make it more intuitive for people? Something like:

```rust
"Vec" => Ok(TokenKind::List), 
```

That way when people see a method that takes a vec, they provide it with a Vec and we do the magic of translating that for them :)

Reply:
That might be a good idea. Will apply.

Reply:
In which case, perhaps we also do the same for `HashMap`, `BTreeMap` etc... Although I'm not so sure about these.

I do like the alias idea, but I'm also torn by keeping the grammar as simple as possible; as this code needs to (eg) be copied onto a Ledger where space is limited.

I'm also happy with supporting Vec but not the others